templates/powershell/build/ListSecrets.ps1

```diff
@@ -0,0 +1,136 @@
+Write-Debug "${functionName}:PSHelperDirectory=$PSHelperDirectory"
+
+try {
+
+    Import-Module $PSHelperDirectory -Force
+    $variablesArray = @()
+    Invoke-CommandLine -Command "az devops configure --defaults organization=$ENV:DevOpOrganization"
+    Invoke-CommandLine -Command "az devops configure --defaults project=$ENV:DevOpsProject"
+    $VariableGroupsArray = $VariableGroups -split ";"
+    if (![string]::IsNullOrEmpty($VarFilter)) {
+        $VarFilterArray = $VarFilter -split ";"
+    } 
+    foreach ($VariableGroup in $VariableGroupsArray) {
+        if ([string]::IsNullOrEmpty($ProgrammeName) -or $VariableGroup -like $ProgrammeName -or $VariableGroup -match $ProgrammeName) {        
+            if ($VariableGroup.Contains('<environment>')) {
+                $VariableGroup = $VariableGroup -replace '<environment>', $EnvName
+            }
+            if ($VariableGroup.Contains($EnvName)) {
+                Write-Host "VariableGroup :$VariableGroup"                  
+                $group = Invoke-CommandLine -Command "az pipelines variable-group list --group-name $VariableGroup --detect true | ConvertFrom-Json"            
+                $groupId = $group.id
+                $variable_group = Invoke-CommandLine -Command "az pipelines variable-group variable list --group-id $groupId --detect true  | ConvertFrom-Json"  
+                $variables = $variable_group.psobject.Properties.Name
+                Write-Host "variables :$variables" 
+                foreach ($variable in $variables) {
+                    if (![string]::IsNullOrEmpty($VarFilterArray)) {
+                        foreach ($filter in $VarFilterArray) {
+                            if ($variable -like $filter -or $variable -match $filter) {      
+                                $variablesArray += $variable             
+                                continue
+                            }
+                            else {
+                                Write-Debug "Variable :$variable does not match with filter :$filter"  
+                            }
+                        }
+                    }
+                    else {                  
+                        $variablesArray += $variable
+                    }
+                }
+                
+                
+            }
+            else {
+                Write-Host "${functionName} :$VariableGroup not related to env: $EnvName"        
+            }
+        }
+        else {
+            Write-Host "VariableGroup :$VariableGroup does not match with ProgrammeName :$ProgrammeName"  
+        }
+    }  
+
+    Write-Host "variablesArray :$variablesArray" 
+    [hashtable]$body = @{}
+    $body.variables = @()
+    foreach ($var in $variablesArray) {
+        $body.variables += @([ordered]@{name = $var; value = "`$($var)" })
+    }
+    $json = $body.variables | ConvertTo-Json -Compress 
+    Write-Host "##vso[task.setvariable variable=secretVariablesJsonObj;]$json"
```

Comment:
I think its a string, can we change the variable name to secretVariablesJson ?

Reply:
updated